File: MQ/rabbitmq-js/consumer.js
const amqp = require('amqplib');
const queue = 'log_queue';

async function consumeMessages() {
  const connection = await amqp.connect('amqp://localhost');
  const channel = await connection.createChannel();
  await channel.assertQueue(queue);

  console.log('Oczekiwanie na wiadomości...');
  channel.consume(queue, msg => {
    if (msg !== null) {
      console.log(`Odebrano: ${msg.content.toString()}`);
      channel.ack(msg);
    }
  });
}

consumeMessages().catch(console.error);

File: MQ/rabbitmq-python/consumer.py
import pika

def callback(ch, method, properties, body):
    print(f"Odebrano: {body.decode()}")
    ch.basic_ack(delivery_tag=method.delivery_tag)

connection = pika.BlockingConnection(pika.ConnectionParameters('localhost'))
channel = connection.channel()

queue_name = 'log_queue'
channel.queue_declare(queue=queue_name, durable=True)

channel.basic_consume(queue=queue_name, on_message_callback=callback)

print('Oczekiwanie na wiadomości...')
channel.start_consuming()
